File: resources/xcode/NativePHP/NativeRender/DividerRenderer.swift
import UIKit

struct DividerViewRenderer: NativeViewRenderer {
    func createView(node: NativeUINode) -> UIView {
        let v = UIView()
        applyProps(v, node: node)
        return v
    }

    func updateView(_ view: UIView, node: NativeUINode) {
        applyProps(view, node: node)
    }

    private func applyProps(_ view: UIView, node: NativeUINode) {
        if let style = node.style, style.borderColor != 0 {
            view.backgroundColor = UIColor(argb: style.borderColor)
        } else {
            view.backgroundColor = UIColor.separator
        }
    }
}

File: resources/androidstudio/app/src/main/java/com/nativephp/mobile/ui/nativerender/DividerRenderer.kt
package com.nativephp.mobile.ui.nativerender

import android.content.Context
import android.graphics.Color
import android.view.View

class DividerViewRenderer : NativeViewRenderer {
    override fun createView(context: Context, node: NativeUINode): View {
        val v = View(context)
        v.setBackgroundColor(Color.parseColor("#E0E0E0"))
        return v
    }

    override fun updateView(view: View, node: NativeUINode) {
        // Divider is just a styled line — color handled by applyStyle or default
    }
}
